翻译2D对象、渲染组件及UI说明四篇 (#1809)
cocos-creator/creator-docs: en/ui-system/components/engine/index.md

# UI Description

## Getting Started with UI

> The difference between defining UI and 2D rendering objects in the engine mainly lies in adaptation and interaction. All UI needs to be under Canvas node to make adaptation behavior, and Canvas component itself inherits from `RenderRoot2D` component, so it can also be used as an entry point for data collection.

Reviewer: ```suggestion
The difference between defining UI and 2D rendering objects in the engine mainly lies in adaptation and interaction. All UI needs to be under a Canvas node to make adaptation behavior, and the Canvas component itself inherits from `RenderRoot2D` component, it can also be used as an entry point for data collection.
```

![resolution-config](resolution_config.png)

Once the design resolution is set, start creating UI elements, all of which are contained under the Canvas node. The Canvas node can be created by clicking the **+** button in the **Hierarchy** panel on the top left and selecting **UI Component -> Canvas**. The Canvas node has a [Canvas](../editor/canvas.md) component, which can be associated with a camera.

> **Notes**:
>
> - Multiple Canvas nodes can exist in a scene, but a Canvas node should not be nested under another Canvas node or its children.
> - Canvas components are not one-to-one with camera, their previous rendering depends on the layer of the node and the Visibility of the camera, so you should pay extra attention to layer management to get the desired rendering effect when you have multiple Canvas.

> Next, you can create UI nodes under the Canvas node. The editor comes with the following UI nodes:

Reviewer: ```suggestion
Next, create UI nodes under the Canvas node. The editor comes with the following UI nodes:
```

![create-ui](./create-ui.png)

UI components can be viewed by selecting the node and clicking **Add Component** in the **Inspector** panel.

![add-ui-component](./add-ui-component.png)

The order in which UI components are rendered is a depth ordering scheme, which means that the ordering of the child nodes under the Canvas node already determines the entire [rendering order](priority.md).

In general game development, the necessary UI elements are not only basic 2D rendering components such as Sprite, Label (text), Mask, but also Layout, Widget (alignment), etc., which are used to quickly build the interface. Sprite and Label are used to render images and text, Mask is mainly used to limit the display content, more commonly used in chat boxes and backpacks, etc. Layout is generally used for single arrangement of buttons, neat arrangement of props in backpacks, etc. <br>

> The last important feature is the Widget, which is mainly used for display alignment. When we finish designing the UI and publish it to different platforms, the actual device resolution of the platform is bound to be different from our design resolution, so we have to make some trade-offs in order to adapt it. We need to add a widget component to it, and always ensure that it is aligned to the top left of our design resolution, see: [alignment strategy](widget-align.md) and [alignment](../editor/widget.md).

Reviewer: ```suggestion
The last important feature is the Widget, which is mainly used for display alignment. When finishing designing the UI and publish it to different platforms, the actual device resolution of the platform is bound to be different from our design resolution, therefore some trade-offs need to be made in order to adapt it. It is necessary to add a widget component to it, and always ensure that it is aligned to the top left of our design resolution. Please review the [alignment strategy](widget-align.md) and [alignment](../editor/widget.md) documentation.
```

> Once the interface is created, some people may notice that the iPhone 7 displays differently than the iPhone X. This is actually the same problem with the device resolution we mentioned above. When you design in design resolution and finally publish in device resolution, there is a pixel deviation because the resolution of different models of mobile devices may not be the same, so there is another conversion process that needs to be done that is screen adaptation. <br>

Reviewer: Do we need this paragraph anymore? These phones are somewhat dated.

Author: @xunyi0 Jasmine问这段要不要删掉，因为里面提到的机型已经很老了。

Reviewer 2: 没事，这里只是举个例子说明不同设备分辨率要怎么做屏幕适配

> As you can see in the **Projects -> Project Settings -> Project Data** page in the menu bar, there are two more options **Fit Width / Fit Height**, which can be easily adapted to different devices by following the screen adaptation rules and combining with the Widget component. The specific adaptation rules can be found in the [Multi-Resolution Adaptation Scheme](multi-resolution.md).

Reviewer: ```suggestion
Notice in the **Projects -> Project Settings -> Project Data** page in the menu bar, there are two more options **Fit Width / Fit Height**, which can be easily adapted to different devices by following the screen adaptation rules and combining with the Widget component. The specific adaptation rules can be found in the [Multi-Resolution Adaptation Scheme](multi-resolution.md) documentation.
```
